src/OnboardingSPA/pages/Steps/DesignHomepageMenu/index.js

```diff
@@ -0,0 +1,175 @@
+import { useViewportMatch } from '@wordpress/compose';
+import { useState, useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+import { getPatterns } from '../../../utils/api/patterns';
+import { getGlobalStyles } from '../../../utils/api/themes';
+import { store as nfdOnboardingStore } from '../../../store';
+import CommonLayout from '../../../components/Layouts/Common';
+import { VIEW_DESIGN_HOMEPAGE_MENU } from '../../../../constants';
+import { LivePreviewSelectableCard } from '../../../components/LivePreview';
+import HeadingWithSubHeading from '../../../components/HeadingWithSubHeading';
+import { useGlobalStylesOutput } from '../../../utils/global-styles/use-global-styles-output';
+
+const StepDesignHomepageMenu = () => {
+
+    const homepagePatternList = [
+        'homepage-1',
+        'homepage-2',
+        'homepage-3'
+    ];
+
+    const homepages_list = {
+        'homepage-1' : [
+            "site-header-left-logo-navigation-inline",
+            "homepage-1",
+            "site-footer"
+        ],
+        'homepage-2' : [
+            "site-header-left-logo-navigation-inline",
+            "homepage-2",
+            "site-footer"
+        ],
+        'homepage-3' : [
+            "site-header-left-logo-navigation-inline",
+            "homepage-3",
+            "site-footer"
+        ],
+    };
+
+    const [isLoaded, setisLoaded] = useState(false);
+    const [globalStyle, setGlobalStyle] = useState();
+    const [homepagePattern, setHomepagePattern] = useState();
+    const [selectedHomepage, setSelectedHomepage] = useState(0);
+
+    const isLargeViewport = useViewportMatch('medium');
+
+    const { currentStep, currentData, storedPreviewSettings } = useSelect(
+        (select) => {
+            return {
+                currentStep: select(nfdOnboardingStore).getCurrentStep(),
+                currentData: select(nfdOnboardingStore).getCurrentOnboardingData(),
+                storedPreviewSettings: select(nfdOnboardingStore).getPreviewSettings()
+            };
+        },
+        []
+    );
+
+    const { setDrawerActiveView, 
+            setIsDrawerOpened, 
+            setIsSidebarOpened, 
+            updatePreviewSettings,
+            setIsDrawerSuppressed, 
+            setCurrentOnboardingData } =
+        useDispatch(nfdOnboardingStore);
+
+    useEffect(() => {
+        if (isLargeViewport) {
+            setIsDrawerOpened(true);
+        }
+        setIsSidebarOpened(false);
+        setIsDrawerSuppressed(false);
+        setDrawerActiveView(VIEW_DESIGN_HOMEPAGE_MENU);
+    }, []);
+
+    function refactorPatterns( homepagePatternData ) {
+
+        var makeHomepagePattern = [];
+
+        for (let key in homepages_list) {
+            var homepage_pattern_array = homepages_list[key];
```

Comment:
```suggestion
            var homepagePatterns = homepages_list[key];
```
Let's stick to camel case here and anywhere else we have used `_`'s